builtin/providers/aws/resource_aws_organization_account.go

```diff
@@ -0,0 +1,230 @@
+		Read:   resourceAwsOrganizationAccountRead,
+		Update: resourceAwsOrganizationAccountUpdate,
+		Delete: resourceAwsOrganizationAccountDelete,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"joined_method": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"joined_timestamp": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringLenBetween(1, 50),
+			},
+			"email": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validateAwsOrganizationAccountEmail,
+			},
+			"iam_user_access_to_billing": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validation.StringInSlice([]string{"ALLOW", "DENY"}, true),
+			},
+			"role_name": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				ValidateFunc: validateAwsOrganizationAccountRoleName,
+			},
+		},
+	}
+}
+
+func resourceAwsOrganizationAccountCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).orgsconn
+
+	// Create the account
+	createOpts := &organizations.CreateAccountInput{
+		AccountName: aws.String(d.Get("name").(string)),
+		Email:       aws.String(d.Get("email").(string)),
+	}
+	if role, ok := d.GetOk("role_name"); ok {
+		createOpts.RoleName = aws.String(role.(string))
+	}
+
+	if iam_user, ok := d.GetOk("iam_user_access_to_billing"); ok {
+		createOpts.IamUserAccessToBilling = aws.String(iam_user.(string))
+	}
+
+	log.Printf("[DEBUG] Account create config: %#v", createOpts)
+
+	var err error
+	var resp *organizations.CreateAccountOutput
+	err = resource.Retry(4*time.Minute, func() *resource.RetryError {
+		resp, err = conn.CreateAccount(createOpts)
+
+		if err != nil {
+			ec2err, ok := err.(awserr.Error)
+			if !ok {
+				return resource.NonRetryableError(err)
+			}
+			if ec2err.Code() == "FinalizingOrganizationException" {
+				log.Printf("[DEBUG] Trying to create account again: %q", ec2err.Message())
+				return resource.RetryableError(err)
+			}
+
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("Error creating account: %s", err)
+	}
+	log.Printf("[DEBUG] Account create response: %#v", resp)
+
+	requestId := *resp.CreateAccountStatus.Id
+
+	// Wait for the account to become available
+	log.Printf("[DEBUG] Waiting for account request (%s) to succeed", requestId)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{"IN_PROGRESS"},
+		Target:       []string{"SUCCEEDED"},
+		Refresh:      resourceAwsOrganizationAccountStateRefreshFunc(conn, requestId),
+		PollInterval: 10 * time.Second,
+		Timeout:      5 * time.Minute,
```

Comment:
So for the account creation we can poll based on the status using `CreateAccountStatus`.  However if you chose to also create an IAM role that takes some time and honestly I had no idea how long.  This was a guess.  During my testing I chose to create an IAM role when creating an account, then assume that role and create some resources in the new account, but it failed because the IAM role didn't yet exist.  I wasn't sure how to check if the new role was created in the new account without being able to assume that role.  
Also, creating an account consumes your email address and you cannot register another account with that same email address.  At one point in my testing I actually ran out of email addresses to use.  I learned later, via emails with AWS support, that you can change your email address associated with the account and then reuse that email address.